msvc/mono.props

```diff
@@ -17,7 +17,8 @@
     <MONO_USE_STATIC_LIBMONO>false</MONO_USE_STATIC_LIBMONO>
   </PropertyGroup>
   <PropertyGroup Label="MonoDirectories">
-    <top_srcdir>$(MSBuildProjectDirectory)/..</top_srcdir>
+    <MonoSourceLocation Condition="'$(MonoSourceLocation)' == '' ">..</MonoSourceLocation>
+    <top_srcdir>$(MSBuildProjectDirectory)/$(MonoSourceLocation)</top_srcdir>
     <MONO_DIR>$(top_srcdir)</MONO_DIR>
```

Comment:
Actually, after trying this with our "client" code that uses `MonoSourceLocation`, I think that I like the existing code better. Current, our client code puts the git clone of the Mono code in a directory named "external\mono" in its parent. So we get a structure like this:

```
external
  |-> mono
platformA
  |-> eglib-platformA.vcxproj
platformB
  |-> eglib-platformB.vcproj
```

Then the file platformA\eglib-platformA.vcxproj can set `MonoSourceLocation` like this:

```
  <PropertyGroup>
    <MonoSourceLocation>..\external\mono</MonoSourceLocation>
  </PropertyGroup
```

If we collapsed everything into `MONO_DIR`, then we would need a leading `\` character in the platformA\eglib-platformA.vcxproj file, like this:

```
  <PropertyGroup>
    <MONO_DIR>\..\external\mono</MONO_DIR>
  </PropertyGroup
```

I think this second case is less read able in the client code, since the need for the `\` character leaks out of mono.props into the client code. So I would prefer to still keep the code in mono.props as-is in this PR.